WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt

```diff
@@ -0,0 +1,114 @@
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.painter.ColorPainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import org.wordpress.android.R
+import org.wordpress.android.ui.compose.theme.AppTheme
+
+private const val DEFAULT_ICON_SIZE = 32
+private const val DEFAULT_ICON_BORDER_WIDTH = 2
+
+// this proportion is calculated based on the Figma design for Jetpack Social (Th1ahHKq53k5JT1PNDMavY-fi-865_13166)
+private const val ICON_OFFSET_PROPORTION = 29f / 36f
+
+/**
+ * This component uses coil's [AsyncImage] internally so it supports any model the regular [AsyncImage] composable can
+ * use. The icons are laid out horizontally, with the first icon being the first element in the list and the last icon
+ * being the last element in the list. The icons are laid out in a way that they overlap each other slightly.
+ *
+ * The space an individual icon occupies is calculated by the sum of the [iconSize] and [iconBorderWidth] parameters.
+ *
+ * @param iconModels a list of models to be used for the icons. The list must have at least 1 element.
+ * @param modifier the modifier to be applied to the layout.
+ * @param iconSize the size of an individual icon.
+ * @param placeholder the placeholder to be used while the icons are loading.
+ */
+@Composable
+fun TrainOfIcons(
+    iconModels: List<Any>,
+    modifier: Modifier = Modifier,
+    contentDescription: String? = null,
+    iconSize: Dp = DEFAULT_ICON_SIZE.dp,
+    iconBorderWidth: Dp = DEFAULT_ICON_BORDER_WIDTH.dp,
+    placeholder: Painter = ColorPainter(colorResource(R.color.placeholder)),
+) {
+    require(iconModels.isNotEmpty()) { "TrainOfIcons must have at least 1 icon" }
+
+    val iconSizeWithBorder = (iconSize.value + 2 * iconBorderWidth.value).toInt()
+
+    Layout(
+        modifier = modifier.then(Modifier.semantics(mergeDescendants = true) {
+            contentDescription?.let {
+                this.contentDescription = it
+            }
+        }),
+        content = {
+            iconModels.forEach { iconModel ->
+                AsyncImage(
+                    model = iconModel,
+                    contentDescription = null,
+                    contentScale = ContentScale.Crop,
+                    placeholder = placeholder,
+                    modifier = Modifier
+                        .size(iconSizeWithBorder.dp)
+                        .clip(CircleShape)
+                        .background(MaterialTheme.colors.surface)
+                        .padding(iconBorderWidth)
+                        .clip(CircleShape)
```

Comment:
Is duplicating the `.clip` needed here?

Reply:
Good point! We actually need them both.

If we remove just the first `.clip`, it looks like this:
<img width="769" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/09074b3e-2d10-4f96-9387-8972fbb40009">

And if we remove just the second one, it looks like this:
<img width="764" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/0cce096c-daa3-4567-b002-206043098cbc">

Reply:
Makes sense 👍 , thanks for elaborating 🙇

Reply:
Just adding more context for what is happening there. The modifier operations run in the order they're called. So, assuming icon size is 32, and border size is 2, what's happening there is:

1. `.size(iconSizeWithBorder.dp)` -> makes the component have size `iconSizeWithBorder`, or `36dp`;
2. `.clip(CircleShape)` -> clip the component to a circle shape, so anything drawn outside of that circle will not show up;
3. `.background(MaterialTheme.colors.surface)` -> color the component background with the `surface` color (e.g.: white for light theme);
4. `.padding(iconBorderWidth)` -> add padding on all sides of size `iconBorderWidth`, or `2dp`, so any content drawn by the component (`AsyncImage`) will be inset on top of the circular `surface` colored background (because of the previous steps), so if the content is opaque it will look like a `2dp` border;
5. `.clip(CircleShape)` finally clips the content that will be drawn, so the loaded image also has a circular shape.

This is why things look like what @RenanLukas showed above:
- If we remove the first `.clip`, then the outside shape will be a square, with `2dp` padding, with a circle image inside.
- If we remove the second `.clip`, then the outside shape will be a circle, with `2dp` padding, with a square image inside.